sdk/core/Azure.Core.Experimental/src/DynamicJson.cs

```diff
@@ -103,6 +104,12 @@ public override string ToString()
             return _element.ToString();
         }
 
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            _element.DisposeRoot();
```

Comment:
There is a challenge here that DynamicJson wraps a MutableJsonElement that may or may not represent the root of the document.  If disposed, it will dispose the full document and not just that element.

We can recommend that users only call Dispose() on a DynamicJson representing the top-level document (e.g. one returned from ToDynamic()), or just make sure it is well-documented that disposing any element in the document disposes the full document.

Reply:
Should we add a test that demonstrates this dispose behavior?